### DISCUSSION TITLE
Original error: Timed out snapshotting com.apple.testmanagerd

### DISCUSSION THREAD

**Discussion by Hamzastreamwide** [appium/appium#16622] [Q&A]
hello , i m struggling to find a solution for this error , it  happened since i upgraded the os to 15.3 it was 14 before and works fine

i add capabilities of snapshottimeout and i increased it and its the same issue 

[debug] [W3C (3f6d60f5)] Calling AppiumDriver.findElement() with args: ["-ios class chain","**/XCUIElementTypeSearchField[`label == \"Search\"`]","3f6d60f5-08db-4639-a269-8160b801c833"]
[debug] [XCUITest] Executing command 'findElement'
[debug] [BaseDriver] Valid locator strategies for this request: xpath, id, name, class name, -ios predicate string, -ios class chain, accessibility id, css selector
[debug] [BaseDriver] Waiting up to 0 ms for condition
[debug] [WD Proxy] Matched '/element' to command name 'findElement'
[debug] [WD Proxy] Proxying [POST /element] to [POST http://127.0.0.1:8300/session/ABB77CD1-6A13-4C8A-894F-0488BA4DE711/element] with body: {"using":"class chain","value":"**/XCUIElementTypeSearchField[`label == \"Search\"`]"}
[debug] [WD Proxy] Got response with status 200: {"value":{"ELEMENT":"4C010000-0000-0000-4226-000000000000","element-6066-11e4-a52e-4f735466cecf":"4C010000-0000-0000-4226-000000000000"},"sessionId":"ABB77CD1-6A13-4C8A-894F-0488BA4DE711"}
[debug] [W3C (3f6d60f5)] Responding to client with driver.findElement() result: {"element-6066-11e4-a52e-4f735466cecf":"4C010000-0000-0000-4226-000000000000","ELEMENT":"4C010000-0000-0000-4226-000000000000"}
[HTTP] <-- POST /wd/hub/session/3f6d60f5-08db-4639-a269-8160b801c833/element 200 2634 ms - 137
[HTTP] 
[HTTP] --> POST /wd/hub/session/3f6d60f5-08db-4639-a269-8160b801c833/element/4C010000-0000-0000-4226-000000000000/value
[HTTP] {"id":"4C010000-0000-0000-4226-000000000000","text":"tunisia","value":["t","u","n","i","s","i","a"]}
[debug] [W3C (3f6d60f5)] Calling AppiumDriver.setValue() with args: [["t","u","n","i","s","i","a"],"4C010000-0000-0000-4226-000000000000","3f6d60f5-08db-4639-a269-8160b801c833"]
[debug] [XCUITest] Executing command 'setValue'
[debug] [WD Proxy] Matched '/element/4C010000-0000-0000-4226-000000000000/value' to command name 'setValue'
[debug] [Protocol Converter] Added 'text' property "tunisia" to 'setValue' request body
[debug] [WD Proxy] Proxying [POST /element/4C010000-0000-0000-4226-000000000000/value] to [POST http://127.0.0.1:8300/session/ABB77CD1-6A13-4C8A-894F-0488BA4DE711/element/4C010000-0000-0000-4226-000000000000/value] with body: {"value":["t","u","n","i","s","i","a"],"text":"tunisia"}
[WD Proxy] Got response with status 404: {"value":{"error":"stale element reference","message":"The previously found element \"\"Search\" SearchField\" is not present in the current view anymore. Make sure the application UI has the expected state. Original error: Timed out snapshotting com.apple.testmanagerd, app is either unresponsive or taking too long to snapshot.","traceback":"(\n\t0   CoreFoundation                      0x00000001805f1110 E2F84645-2905-31EF-8EC7-3CC19C3CDDB3 + 626960\n\t1   libobjc.A.dylib                     0x0000000198e41d64 objc_exception_throw + 60\n\t2   WebDriverAgentLib                   0x000000010ac9c068 -[XCUIElement(FBUtilities) fb_takeSnapshot] + 900\n\t3   WebDriverAgentLib                   0x000000010aca30dc -[XCUIElement(FBTyping) fb_prepareForTextInputWithSnapshot:] + 288\n\t4   WebDriverAgentLib                   0x000000010aca330c -[XCUIElement(FBTyping) fb_typeText:shouldClear:frequency:error:] + 320\n\t5   WebDriverAgentLib                   0x000000010acd7aa0 +[FBElementCommands handleSetValue:] + 126...
[debug] [W3C] Matched W3C error code 'stale element reference' to StaleElementReferenceError
[debug] [W3C (3f6d60f5)] Encountered internal error running command: StaleElementReferenceError: The previously found element ""Search" SearchField" is not present in the current view anymore. Make sure the application UI has the expected state. Original error: Timed out snapshotting com.apple.testmanagerd, app is either unresponsive or taking too long to snapshot.
[debug] [W3C (3f6d60f5)]     at errorFromW3CJsonCode (/usr/local/lib/node_modules/appium/node_modules/appium-base-driver/lib/protocol/errors.js:780:25)
[debug] [W3C (3f6d60f5)]     at ProxyRequestError.getActualError (/usr/local/lib/node_modules/appium/node_modules/appium-base-driver/lib/protocol/errors.js:663:14)
[debug] [W3C (3f6d60f5)]     at JWProxy.command (/usr/local/lib/node_modules/appium/node_modules/appium-base-driver/lib/jsonwp-proxy/proxy.js:272:19)
[debug] [W3C (3f6d60f5)]     at runMicrotasks (<anonymous>)
[debug] [W3C (3f6d60f5)]     at processTicksAndRejections (internal/process/task_queues.js:93:5)
[debug] [W3C (3f6d60f5)]     at XCUITestDriver.proxyCommand (/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/lib/commands/proxy-helper.js:96:12)
[debug] [W3C (3f6d60f5)]     at XCUITestDriver.setValue (/usr/local/lib/node_modules/appium/node_modules/appium-xcuitest-driver/lib/commands/element.js:192:5)
[HTTP] <-- POST /wd/hub/session/3f6d60f5-08db-4639-a269-8160b801c833/element/4C010000-0000-0000-4226-000000000000/value 404 2773 ms - 1432
[HTTP] 

org.openqa.selenium.StaleElementReferenceException: The previously found element ""Search" SearchField" is not present in the current view anymore. Make sure the application UI has the expected state. Original error: Timed out snapshotting com.apple.testmanagerd, app is either unresponsive or taking too long to snapshot.
For documentation on this error, please visit: https://www.seleniumhq.org/exceptions/stale_element_reference.html
Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
System info: host: 'MacBook-Pro-Jenkins.local', ip: 'fe80:0:0:0:445:f11d:3f73:8df7%en0', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '11.3', java.version: '11.0.12'
Driver info: io.appium.java_client.ios.IOSDriver
Capabilities {app: /Users/streamwide/.jenkins/..., automationName: XCUITest, browserName: , databaseEnabled: false, deviceName: iPhone, forceAppLaunch: true, fullReset: false, javascriptEnabled: true, launchTimeout: 500000, locationContextEnabled: false, networkConnectionEnabled: false, newCommandTimeout: 0, noReset: true, platform: MAC, platformName: IOS, platformVersion: 12, realDeviceScreenshotter: idevicescreenshot, screenshotWaitTimeout: 60, shouldTerminateApp: true, showXcodeLog: true, simpleIsVisibleCheck: true, takesScreenshot: true, udid: 00008020-00122C222228002E, useJSONSource: true, useNewWDA: false, usePrebuiltWDA: true, waitForIdleTimeout: 0, waitForQuiescence: false, wdaLocalPort: 8300, webStorageEnabled: false, xcodeOrgId: 36WZCVM7SX, xcodeSigningId: Apple Development, –session-override: true}
Session ID: 3f6d60f5-08db-4639-a269-8160b801c833

**Comment by Hamzastreamwide** [comment]
this have been changed after the reload {"using":"class chain","value":"/XCUIElementTypeSearchField[label == \"Search\"]"} its becomes {"using":"class chain","value":"/XCUIElementTypeSearchField[label == \"Search\"][1]"}

**Reply by KazuCocoa** [reply]
`customSnapshotTimeout` might help https://github.com/appium/appium-xcuitest-driver#settings-api
The root cause is the snapshot by testmanagerd (by Apple) responded slowly. As Appium, it only can wait for the response longer by the settings api, but the result depends on the testmanagerd.
